ipfx/nwb_writer.py

```diff
@@ -0,0 +1,71 @@
+import pynwb
+from allensdk.core.nwb_data_set import NwbDataSet
+import ipfx.nwb_reader as nwb_reader
+
+
+class NwbWriter(object):
+
+    def __init__(self, nwb_file_name):
+        self.nwb_file_name = nwb_file_name
+
+    def add_spike_times(self, sweep_num, spike_times):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+
+class Nwb1Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+        self.nwbfile = NwbDataSet(self.nwb_file_name)
+
+    def add_spike_times(self, sweep_num, spike_times):
+        self.nwbfile.set_spike_times(sweep_num, spike_times)
+
+    def save(self):
+        pass
+
+
+class Nwb2Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+
+        io = pynwb.NWBHDF5IO(self.nwb_file_name, 'a')
+        self.nwbfile = io.read()
+        io.close()
+
+        self.nwbfile.add_unit_column('sweep_number', description="sweep number")
+
+    def add_spike_times(self, sweep_num, spike_times):
+        self.nwbfile.add_unit(spike_times=spike_times, sweep_number=sweep_num)
```

Comment:
the units table [has a bunch of fields](https://github.com/NeurodataWithoutBorders/pynwb/blob/f62f52e300a098b161ccca7c1c5fd598feef28e5/src/pynwb/misc.py#L160). Are you not using those?

Not saying this is a bad decision, mind.

Reply:
Yes, I am not using those.